webapp/src/ts/modals/session-expired/session-expired.component.ts

```diff
@@ -0,0 +1,24 @@
+import { MmModalAbstract } from '../mm-modal/mm-modal';
+import { SessionService } from '@mm-services/session.service';
+
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import {Component} from '@angular/core';
+
+
+@Component({
+  selector: 'session-expired',
+  templateUrl: './session-expired.component.html'
+})
+export class SessionExpiredComponent extends MmModalAbstract {
+
```

Comment:
Adding an id here should guarantee that you're not going to be shown multiple modals, please take a look at the reloading modal (which has the same problem, where it can be triggered by multiple events in quick succession).